File: depense/api/src/main/java/org/cocktail/gfc/depense/api/ServiceFaitCommand.java
package org.cocktail.gfc.depense.api;

import org.cocktail.gfc.common.configuration.internal.immutable.ImmutableStyle;
import org.immutables.value.Value;

public interface ServiceFaitCommand {

    @Value.Immutable
    @ImmutableStyle
    public interface AbstractCreerSfCommand extends ServiceFaitCommand {
      @Value.Parameter
      Long getEjId();
      @Value.Parameter      
      Integer getExeOrdre();
      @Value.Parameter
      String getReference();
      @Value.Parameter
      String getCommentaire();
      @Value.Parameter
      String getDateSf();
      @Value.Parameter
      Long getPersId();
      @Value.Parameter
      String getNomApplication();
      @Value.Parameter
      String getLignesSf();
    }
    
    @Value.Immutable
    @ImmutableStyle
    public interface AbstractMajSfCommand extends ServiceFaitCommand {
      @Value.Parameter      
      Long getExeOrdre();
      @Value.Parameter
      String getReference();
      @Value.Parameter
      String getCommentaire();
      @Value.Parameter
      String getDateSf();
      @Value.Parameter
      Long getPersId();
      @Value.Parameter
      String getNomApplication();
      @Value.Parameter
      String getLignesSf();
    }
    
}

File: depense/api/src/main/java/org/cocktail/gfc/depense/api/ServiceFaitCommand.kt
package org.cocktail.gfc.depense.api;

interface ServiceFaitCommand {
}

data class CreerSfCommand(
      val ejId: Long,
      val exeOrdre: Integer,
      val reference: String,
      val commentaire: String,
      val dateSf: String,
      val persId: Long,
      val nomApplication: String,
      val lignesSf: String
): ServiceFaitCommand

data class MajSfCommand(
        val exeOrdre: Integer,
        val reference: String,
        val commentaire: String,
        val dateSf: String,
        val persId: Long,
        val nomApplication: String,
        val lignesSf: String
): ServiceFaitCommand

